Compatibility: test_external_data_source_secret fails with ModuleNotFoundError: No module named 'ydb.tests.compatibility'

## Summary

All 10 test chunks of `test_external_data_source_secret.py` in `federated_queries` fail to collect tests due to a Python import error: `ModuleNotFoundError: No module named 'ydb.tests.compatibility'`. The tests are currently MUTED.

## Environment

- **Workflow:** [Regression-run_compatibility #537](https://github.com/ydb-platform/ydb/actions/runs/25760246420)
- **Build Preset:** `relwithdebinfo`
- **Branch:** `main`
- **Commit:** `5c8005d`
- **Versions under test:**
  - stable-26-1-1 (`1bf5f58da57b531b2937fd68be6565a8ce8f8792`)
  - prestable-26-2 (`acefcd6b8b05ac84d99a3679c3a7b9287a609bfb`)
  - prestable-26-3 (`3590e0328592b126614896e1451994b167cb315f`)

## HTML Reports

- [Try 1](https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_compatibility/25760246420/ya-main-x86-64/try_1/ya-test.html#MUTE)
- [Try 2](https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_compatibility/25760246420/ya-main-x86-64/try_2/ya-test.html#MUTE)
- [Try 3](https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_compatibility/25760246420/ya-main-x86-64/try_3/ya-test.html#MUTE)

## Affected Tests (MUTED, all 3 tries, 10 chunks)

- `federated_queries/py3test.[test_external_data_source_secret.py 0/10] chunk`
- `federated_queries/py3test.[test_external_data_source_secret.py 1/10] chunk`
- `federated_queries/py3test.[test_external_data_source_secret.py 2/10] chunk`
- `federated_queries/py3test.[test_external_data_source_secret.py 3/10] chunk`
- `federated_queries/py3test.[test_external_data_source_secret.py 4/10] chunk`
- `federated_queries/py3test.[test_external_data_source_secret.py 5/10] chunk`
- `federated_queries/py3test.[test_external_data_source_secret.py 6/10] chunk`
- `federated_queries/py3test.[test_external_data_source_secret.py 7/10] chunk`
- `federated_queries/py3test.[test_external_data_source_secret.py 8/10] chunk`
- `federated_queries/py3test.[test_external_data_source_secret.py 9/10] chunk`

## Error

```
library/python/pytest/plugins/collection.py:78: in _getobj
    __import__(self.ya_module_name)
library/python/pytest/rewrite.py:110: in exec_module
    self.load_module(module)
library/python/pytest/rewrite.py:101: in load_module
    exec(co, module.__dict__)  # noqa
ydb/tests/compatibility/federated_queries/test_external_data_source_secret.py:5: in <module>
    from ydb.tests.compatibility.federated_queries.test_external_data_source import ExternalDataTableTestBase
E   ModuleNotFoundError: No module named 'ydb.tests.compatibility'

During handling of the above exception, another exception occurred:
library/python/pytest/plugins/collection.py:93: in _getobj
    six.reraise(etype, type(exc)('{}\n{}'.format(exc, msg)), tb)
E   ModuleNotFoundError: No module named 'ydb.tests.compatibility'
E   Failed to load module "test_external_data_source_secret" and obtain list of tests due to an error
```

## Reproduction

```bash
./ya make -tA --build relwithdebinfo \
  -DYDB_COMPAT_INIT_REF=stable-26-1-1 \
  -DYDB_COMPAT_INTER_REF=prestable-26-2 \
  -DYDB_COMPAT_TARGET_REF=prestable-26-3 \
  ydb/tests/compatibility/federated_queries
```

## Notes

- All 10 test chunks fail to collect tests due to a Python import error.
- The module `ydb.tests.compatibility` is not found, suggesting a missing `__init__.py` or incorrect PYTHONPATH configuration in the test environment.
- The import chain starts at `test_external_data_source_secret.py:5`, which imports `ExternalDataTableTestBase` from `test_external_data_source.py` using an absolute import path.
- This is a build/packaging issue rather than a test logic bug.

--- a/ydb/tests/compatibility/federated_queries/test_external_data_source_secret.py
+++ b/ydb/tests/compatibility/federated_queries/test_external_data_source_secret.py
@@ -2,7 +2,7 @@ import pytest
 
 from ydb.tests.oss.ydb_sdk_import import ydb
 from ydb.tests.library.compatibility.fixtures import MixedClusterFixture, RestartToAnotherVersionFixture, RollingUpgradeAndDowngradeFixture
-from ydb.tests.compatibility.federated_queries.test_external_data_source import ExternalDataTableTestBase
+from test_external_data_source import ExternalDataTableTestBase
 
 
 class ExternalDataSourceSecretCompatibilityBase(ExternalDataTableTestBase):
@@ -13,10 +13,15 @@ class ExternalDataSourceSecretCompatibilityBase(ExternalDataTableTestBase):
             query = """
                 DROP OBJECT IF EXISTS s3_access_key (TYPE SECRET);
                 DROP OBJECT IF EXISTS s3_secret_key (TYPE SECRET);
-                DROP SECRET IF EXISTS `s3_access_key`;
-                DROP SECRET IF EXISTS `s3_secret_key`;
             """
             session_pool.execute_with_retries(query)
+
+            # DROP SECRET does not support IF EXISTS in the grammar; tolerate "not exists" errors.
+            for secret_name in ("s3_access_key", "s3_secret_key"):
+                try:
+                    session_pool.execute_with_retries(f"DROP SECRET `{secret_name}`;")
+                except ydb.issues.Error:
+                    pass
 
             if kind == "object":
                 query = f"""
@@ -39,16 +44,30 @@ class ExternalDataSourceSecretCompatibilityBase(ExternalDataTableTestBase):
 
             session_pool.execute_with_retries(query)
 
-            query = f"""
-                CREATE EXTERNAL DATA SOURCE s3_source WITH (
-                    SOURCE_TYPE = "ObjectStorage",
-                    LOCATION = "{s3_endpoint}/{s3_bucket}",
-                    AUTH_METHOD="AWS",
-                    AWS_ACCESS_KEY_ID_SECRET_NAME="s3_access_key",
-                    AWS_SECRET_ACCESS_KEY_SECRET_NAME="s3_secret_key",
-                    AWS_REGION="us-east-1"
-                );
-            """
+            if kind == "object":
+                query = f"""
+                    CREATE EXTERNAL DATA SOURCE s3_source WITH (
+                        SOURCE_TYPE = "ObjectStorage",
+                        LOCATION = "{s3_endpoint}/{s3_bucket}",
+                        AUTH_METHOD="AWS",
+                        AWS_ACCESS_KEY_ID_SECRET_NAME="s3_access_key",
+                        AWS_SECRET_ACCESS_KEY_SECRET_NAME="s3_secret_key",
+                        AWS_REGION="us-east-1"
+                    );
+                """
+            elif kind == "schema":
+                query = f"""
+                    CREATE EXTERNAL DATA SOURCE s3_source WITH (
+                        SOURCE_TYPE = "ObjectStorage",
+                        LOCATION = "{s3_endpoint}/{s3_bucket}",
+                        AUTH_METHOD="AWS",
+                        AWS_ACCESS_KEY_ID_SECRET_PATH="s3_access_key",
+                        AWS_SECRET_ACCESS_KEY_SECRET_PATH="s3_secret_key",
+                        AWS_REGION="us-east-1"
+                    );
+                """
+            else:
+                raise ValueError(f"Unknown secret kind: {kind}")
 
             session_pool.execute_with_retries(query)
 

